### PR TITLE
Test Type C deconflicting which is the reverse of Type B

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: test/Avro/DeconflictSpec.hs [hunk 1]
```diff
@@ -92,3 +94,34 @@ spec = describe "Avro.DeconflictSpec" $ do
       let Right deconflicted = A.deconflict BW.schema'Foo BR.schema'Foo decodedAvro
 
       A.fromAvro deconflicted `shouldBe` Success BR.sampleValue
+
+  describe "Type C" $ do
```
Comment (dsturnbull) [inline test/Avro/DeconflictSpec.hs]: The name should say what it's doing I think.

File: test/Avro/DeconflictSpec.hs [hunk 2]
```diff
@@ -92,3 +94,34 @@ spec = describe "Avro.DeconflictSpec" $ do
...
+      let res = AL.deconflict CW.schema'Foo CR.schema'Foo decodedAvro
+
+      AL.fromLazyAvro res `shouldBe` Success CR.sampleValue
+
+    it "should deconflict lazy container" $ do
+      w <- liftIO $ A.encodeContainer [[ CW.sampleValue ]]
+      AL.decodeContainer w `shouldBe` [ Right CR.sampleValue ]
+
+    it "should deconflict lazy value" $ do
+      let payload = A.encode AW.sampleValue
```
Comment (AlexeyRaga) [inline test/Avro/DeconflictSpec.hs]: Probably `Type A` test does not belong here?

**Comment by AlexeyRaga** [comment]
Isn't `C.Reader` == `B.Writer` and `C.Writer` == `B.Writer`? 
Can't we reuse `B`?

File: test/Avro/DeconflictSpec.hs (hunk 1 above)
Comment (AlexeyRaga) [reply test/Avro/DeconflictSpec.hs]: It is just a group... Names in `it` are saying something.

**Pull request merged**